xarray/core/combine.py

```diff
@@ -620,21 +581,142 @@ def auto_combine(datasets, concat_dim=_CONCAT_DIM_DEFAULT,
     --------
     concat
     Dataset.merge
-    """  # noqa
-
-    # Coerce 1D input into ND to maintain backwards-compatible API until API
-    # for N-D combine decided
-    # (see https://github.com/pydata/xarray/pull/2553/#issuecomment-445892746)
-    if concat_dim is None or concat_dim == _CONCAT_DIM_DEFAULT:
-        concat_dims = concat_dim
-    elif not isinstance(concat_dim, list):
-        concat_dims = [concat_dim]
+    """
+
+    message = """In xarray version 0.14 `auto_combine` will be deprecated."""
+
+    if concat_dim is '_not_supplied':
```

Comment:
I should have caught this before, but please always use `==` for string comparisons.